Include/internal/pycore_code.h

```diff
@@ -295,31 +295,32 @@ saturating_start(void)
 }
 
 static inline void
-record_cache_hit(_PyAdaptiveEntry *entry) {
```

Comment:
Could you revert this change? Most, if not all, other instruction families will use `_PyAdaptiveEntry`, so I think it helps readability to leave this as it is.
Rather than exposing the `counter` field of `_PyAdaptiveEntry`, you could use `saturating_increment` in `MISS_WITH_OPARG_COUNTER`